lib/coreboot/cbmem_console.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <coreboot/cbmem_console.h>
+#include <xlat_tables_v2.h>
+
+/*
+ * The console is initially validated by coreboot, so we can trust it. But at
+ * runtime the non-secure OS could manipulate the size field to trick us into
+ * making bad memory accesses. Therefore we need to store the known-good size
```

Comment:
Yeah, I know. I'm pretty confident that this is bullet-proof but happy to get more eyes on the code, of course. (Only coreboot_cbmc_putc examines the contents of the buffer at runtime, and it's checking the address it writes to ensure that it cannot lie outside the buffer.)